src/traces/sankey/render.js

```diff
@@ -155,7 +155,7 @@ function linkModel(uniqueKeys, d, l) {
     var tc = tinycolor(l.color);
     var basicKey = l.source.label + '|' + l.target.label;
     var foundKey = uniqueKeys[basicKey];
-    uniqueKeys[basicKey] = (foundKey === void(0) ? foundKey : 0) + 1;
+    uniqueKeys[basicKey] = (foundKey === void(0) ? 0 : foundKey) + 1;
```

Comment:
The code relies on a unique key for the purpose of entering / exiting the proper links. A simple solution would be to lean on the index for this. But it's better to associate semantically meaningful elements, e.g. some specific node or link, with the DOM element - useful for future possible animations or tweening. As we don't require an unique identifier array from users, an approximation is

- for nodes: the node label
- for links: the label of the nodes they link

As there can be more than one link between two arbitrary nodes, the data binding key is made unique by a lookup that generates a new key for additional links. The ternary got flipped during an unrelated refactoring and it only came up with 4+ links.